internal/cli/lib/search.go

```diff
@@ -154,33 +155,36 @@ func (res result) Data() interface{} {
 		}
 
 		names := []LibName{}
-		results := res.results.GetLibraries()
-		for _, lib := range results {
+		for _, lib := range res.results.Libraries {
+			if lib == nil {
+				continue
+			}
 			names = append(names, LibName{lib.Name})
 		}
 
-		return NamesOnly{
-			names,
-		}
+		return NamesOnly{names}
 	}
 
 	return res.results
 }
 
-func (res result) String() string {
-	results := res.results.GetLibraries()
+func (res librarySearchResult) String() string {
+	results := res.results.Libraries
 	if len(results) == 0 {
 		return tr("No libraries matching your search.")
 	}
 
 	var out strings.Builder
 
-	if res.results.GetStatus() == rpc.LibrarySearchStatus_LIBRARY_SEARCH_STATUS_FAILED {
+	if string(res.results.Status) == rpc.LibrarySearchStatus_name[int32(rpc.LibrarySearchStatus_LIBRARY_SEARCH_STATUS_FAILED)] {
```

Comment:
Ouch...
Maybe this way:
```suggestion
	if string(res.results.Status) == rpc.LibrarySearchStatus_LIBRARY_SEARCH_STATUS_FAILED.String() {
```

BTW I'm not sure this is the result I want:

```
Nome                                     Installato Disponibile Posizione             Descrizione
Adafruit ADXL343                         1.2.0      1.6.2       LIBRARY_LOCATION_USER Unified driver for the ADXL343 Accele...
Adafruit APDS9960 Library                1.1.1      1.2.3       LIBRARY_LOCATION_USER This is a library for the Adafruit AP...
```

I'd like to have a user-readable string like:
```
Nome                                     Installato Disponibile Posizione  Descrizione
Adafruit ADXL343                         1.2.0      1.6.2       user dir   Unified driver for the ADXL343 Accele...
Adafruit APDS9960 Library                1.1.1      1.2.3       user dir   This is a library for the Adafruit AP...
```

and the same in the JSON:
```
      "location": "user dir",
      "layout": "recursive",
```
instead of
```
      "location": "LIBRARY_LOCATION_USER",
      "layout": "LIBRARY_LAYOUT_RECURSIVE",
```

To do that I think we must recreate the enum in the `result` package as we did for the rpc structs...
Let's chat about this next week.